src/Traits/SearchResultHelper.php

```diff
@@ -57,8 +67,11 @@ protected function fetchProducts(Criteria $criteria, SalesChannelContext $contex
     private function fixResultOrder(EntitySearchResult $result, Criteria $criteria): EntitySearchResult
     {
         $sortedElements = $this->sortElementsByIdArray($result->getElements(), $criteria->getIds());
-        $result->clear();
+        if (!$sortedElements) {
```

Comment:
If there is nothing to sort, the method `fixResultOrder` shouldn't even be called.